toolchain/check/pattern_match.cpp

```diff
@@ -280,10 +271,14 @@ auto MatchContext::EmitPatternMatch(Context& context,
     }
     case CARBON_KIND(SemIR::ReturnSlotPattern return_slot_pattern): {
       CARBON_CHECK(kind_ == MatchKind::Callee);
-      return_slot_id_ = context.AddInst<SemIR::ReturnSlot>(
+      auto return_slot_id = context.AddInst<SemIR::ReturnSlot>(
```

Comment:
For my own education, why does this create a ReturnSlot id here just to check that it's not valid?

Reply:
`LookupOrAddName` has kind of an odd API: it looks up the given name and returns the result on success, or else adds the name to name lookup and returns an invalid inst. So the code below is verifying that this name wasn't already present in lookup.